sys/net/application_layer/dhcpv6/_dhcpv6.h

```diff
@@ -0,0 +1,226 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @name    DHCPv6 multicast addresses
+ * @see [draft-ietf-dhc-rfc3315bis-12, section 7.1]
+ *      (https://tools.ietf.org/html/draft-ietf-dhc-rfc3315bis-12#section-7.1)
+ * @{
+ */
+/**
+ * @brief   Multicast address used by clients to communicate with neighboring
+ *          relay agents and servers
+ *
+ * @note    Corresponds with `All_DHCP_Relay_Agents_and_Servers` in the draft.
+ *
+ */
+#define DHCPV6_ALL_RELAY_AGENTS_AND_SERVERS { 0xff, 0x02, 0x00, 0x00, \
+                                              0x00, 0x00, 0x00, 0x00, \
+                                              0x00, 0x00, 0x00, 0x00, \
+                                              0x00, 0x01, 0x00, 0x02  \
+                                            }
+/** @} */
+
+/**
+ * @name    DHCPv6 transmission and retransmission parameters
+ * @see [draft-ietf-dhc-rfc3315bis-12, section 7.6]
+ *      (https://tools.ietf.org/html/draft-ietf-dhc-rfc3315bis-12#section-7.6)
+ * @{
+ */
+#define DHCPV6_SOL_MAX_DELAY        (1U)    /**< SOL_MAX_DELAY (in sec) */
+#define DHCPV6_SOL_TIMEOUT          (1U)    /**< SOL_TIMEOUT (in sec) */
+#define DHCPV6_SOL_MAX_RT           (3600U) /**< SOL_MAX_RT (in sec) */
+
+#define DHCPV6_REQ_TIMEOUT          (1U)    /**< REQ_TIMEOUT (in sec) */
+#define DHCPV6_REQ_MAX_RT           (30U)   /**< REQ_MAX_RT (in sec) */
+#define DHCPV6_REQ_MAX_RC           (10U)   /**< REQ_MAX_RC */
+
+#define DHCPV6_REN_TIMEOUT          (10U)   /**< REN_TIMEOUT (in sec) */
+#define DHCPV6_REN_MAX_RT           (600U)  /**< REN_MAX_RT (in sec) */
+
+#define DHCPV6_REB_TIMEOUT          (10U)   /**< REB_TIMEOUT (in sec) */
+#define DHCPV6_REB_MAX_RT           (600U)  /**< REB_MAX_RT (in sec) */
+/** @} */
+
+#define DHCPV6_DUID_MAX_LEN         (128U)  /**< maximum length of DUID */
+
+/**
+ * @name DHCPv6 message formats
+ * @{
+ */
+/**
+ * @brief   Client/Server message header
+ * @see [draft-ietf-dhc-rfc3315bis-12, section 8]
+ *      (https://tools.ietf.org/html/draft-ietf-dhc-rfc3315bis-12#section-8)
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t type;   /**< message type (see [DHCPv6 messeg types ](@ref net_dhcp6_msg_types)) */
+    uint8_t tid[3]; /**< transaction ID */
+} dhcpv6_msg_t;
+/** @} */
+
+/**
+ * @brief   Generic storage DUID
+ */
+typedef union {
+    uint8_t u8[DHCPV6_DUID_MAX_LEN];    /**< array representation */
+    dhcpv6_duid_l2_t duid_l2;           /**< DUID-L2 type */
+} dhcpv6_duid_t;
+/** @} */
+
+/**
+ * @name DHCPv6 options
+ * @{
+ */
+/**
+ * @brief   General DHCPv6 option format
+ * @see [draft-ietf-dhc-rfc3315bis-12, section 21.1]
+ *      (https://tools.ietf.org/html/draft-ietf-dhc-rfc3315bis-12#section-21.1)
+ */
+typedef struct __attribute__((packed)) {
+    network_uint16_t type;      /**< [option code](@ref net_dhcp6_opt_codes) */
+    network_uint16_t len;       /**< length of dhcpv6_opt_t::data in byte */
+    uint8_t data[];             /**< option data */
```

Comment:
I guess this also would cause the problem lined out in #8767. I'll address as soon as I find the time.